sycl/include/sycl/types.hpp

```diff
@@ -743,7 +747,8 @@ template <typename Type, int NumElements> class vec {
   template <size_t... Is>
   constexpr vec(const std::array<vec_data_t<DataT>, NumElements> &Arr,
                 std::index_sequence<Is...>)
-      : m_Data{vec_data_t<DataT>(static_cast<DataT>(Arr[Is]))...} {}
+      : m_Data{
+        vec_data_t<DataT>( static_cast<DataT>(Arr[Is]) )...} {}
```

Comment:
This is the variadic constructor where the float() operator is getting called for bfloat16, but not for sycl::half.